plugins/scaffolder-backend/src/ScaffolderPlugin.ts

```diff
@@ -104,6 +106,26 @@ export const scaffolderPlugin = createBackendPlugin({
           }),
         ];
 
+        lifecycle.addShutdownHook(async () => {
```

Comment:
Oh hang on. I was very unclear, I realize. What I had in mind was not to cancel all global tasks. Remember that this is a cluster of collaborative machines. If you shut down one single host you don't want the entire cluster to cancel everything it's doing globally, even on hosts that are NOT being shut down.

What I think is useful, is to cleanly abort all tasks that _this particular host_ is working on, by canceling the `cancelSignal` of all of the workflow runners and marking them as canceled

Reply:
Oh yeah, that's a very good point

Reply:
@freben I have no doubts of implementing the part with shutdown hook, and just to keep this PR with configurable heartbeats.

Though the idea is great, but there are several things:

* the process when task is terminated is very unpredictable, for me locally it just terminates at some random place and doesn't wait till the full flow to end
* in real Kubernetes when pod goes to a terminating state, any calls to the database are not reachable anymore (at least that works like that in GKE, I was playing with that)
* I'm working on Mac, and *addShutdownHook* actually doesn't work, how it described. I can only make it triggered when I change the code, and hot-reload triggers it. Ctrl-C, Command-C, Ctrl-D, whatever else doesn't trigger this hook.

Reply:
Alright, let's maybe leave that out then